x509util/certcheck/certcheck.go

```diff
@@ -131,34 +139,46 @@ func chainFromSite(target string) ([]*x509.Certificate, error) {
 
 	// Convert base crypto/x509.Certificates to our forked x509.Certificate type.
 	goChain := conn.ConnectionState().PeerCertificates
+	var nfe *x509.NonFatalErrors
 	chain := make([]*x509.Certificate, len(goChain))
 	for i, goCert := range goChain {
 		cert, err := x509.ParseCertificate(goCert.Raw)
-		if err != nil {
+		if x509.IsFatal(err) {
 			return nil, fmt.Errorf("%s: failed to convert Go Certificate [%d]: %v", target, i, err)
+		} else if err, ok := err.(x509.NonFatalErrors); ok {
+			nfe = nfe.Append(&err)
 		}
```

Comment:
nit: It'd be safest to have a catch-all at the end of this if-else in case `err != nil` but it's not of an expected type (right now, it'll get silently ignored).

Reply:
Done (but that shouldn't ever happen).